back/src/main/java/ro/utcn/sd/he/assignment1/controller/QuestionsController.java

```diff
@@ -0,0 +1,55 @@
+package ro.utcn.sd.he.assignment1.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+import ro.utcn.sd.he.assignment1.command.AddQuestionCommand;
+import ro.utcn.sd.he.assignment1.command.Command;
+import ro.utcn.sd.he.assignment1.dto.QuestionDTO;
+import ro.utcn.sd.he.assignment1.dto.UserDTO;
+import ro.utcn.sd.he.assignment1.model.Question;
+import ro.utcn.sd.he.assignment1.service.QuestionService;
+import ro.utcn.sd.he.assignment1.service.QuestionTagService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RequiredArgsConstructor
+@RestController
+public class QuestionsController {
+    private final QuestionService questionService;
+    private final QuestionTagService questionTagService;
+    private List<Command> commands = new ArrayList<>();
+    @GetMapping("questions/{id}")
+    public UserDTO readSingle(@PathVariable int id) {
+        //TODO
+        return null;
+    }
+
+    @GetMapping("/questions")
+    public List<Question> readAll() {
+        return questionService.listQuestions();
+    }
+
+    @GetMapping("/questions/title/{title}")
```

Comment:
For a better REST API, I would suggest using `@RequestParams` instead of `@PathVariables` for the filtering stuff. In the end, for a REST perspective, all three query endpoints (findAll, findByTag and findByTitle) are just one big query endpoint with different filter criteria.